src/Button/Button.component.tsx

```diff
@@ -3,113 +3,602 @@ import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
 import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
-import { darken } from 'polished';
+import { transparentize } from 'polished';
 // THEME
 import { colors, fonts, sizes } from '../theme';
+import { Colors } from '../Typography/Typography.component';
+
+type Variant = 'primary' | 'outline' | 'minimal';
+
+type ButtonSize = 'xlarge' | 'large' | 'small' | 'xsmall';
+
+interface Theme {
+    base: string;
+    light: string;
+    dark: string;
+}
 
 interface ButtonProps {
     children?: any;
     disabled?: boolean;
+    revealed?: boolean;
     showSpinner?: boolean;
     className?: string;
 
+    flip?: boolean;
+    colorTheme?: Theme | Colors;
+    minWidth?: string;
+    block?: boolean;
+    icon?: any;
```

Comment:
We should export a const from icons with all the names of the icons. This has become a pretty common sticking point in other places. Stop by my desk and I'll show you how to generate that in like 2s.

Reply:
paste list here.

Reply:
```type AnchorIcons = 'AddEvent' | 'ArrowBack' | 'ArrowForward' | 'AvatarIcon' | 'AvatarOutline' | 'BarCode' | 'BulletList' | 'Calendar' | 'Camera' | 'Cart' | 'CashBack' | 'Cells' | 'Chat' | 'Check' | 'CheckSmall' | 'ChevronDown' | 'ChevronDownSmall' | 'ChevronLeft' | 'ChevronLeftSmall' | 'ChevronRight' | 'ChevronRightSmall' | 'ChevronUp' | 'ChevronUpSmall' | 'Clock' | 'Close' | 'CloseSmall' | 'CommentMore' | 'CreditCard' | 'CrossHairs' | 'Cut' | 'Disabled' | 'Dislike' | 'Download' | 'Ellipses' | 'EllipsesVertical' | 'Envelope' | 'EnvelopeOpen' | 'Error' | 'Expand' | 'Gear' | 'GiftCard' | 'Hamburger' | 'Heart' | 'HeartOutline' | 'Home' | 'Info' | 'InfoOutline' | 'Laptop' | 'Lightning' | 'Like' | 'ListIcon' | 'Lock' | 'Map' | 'Marker' | 'MarkerOutline' | 'Mobile' | 'News' | 'Pencil' | 'Play' | 'Plus' | 'PlusSmall' | 'Print' | 'Question' | 'QuestionOutline' | 'Refresh' | 'RetailMeNotLogo' | 'SadFace' | 'Search' | 'Share' | 'Sliders' | 'Star' | 'StarHalf' | 'StarOutline' | 'Success' | 'SuccessOutline' | 'Tag' | 'TagAdd' | 'Upload';```

Reply:
atm it takes a react element, not the name. e.g.
```jsx
<Button icon={Close} />
```
not
```jsx
<Button icon="Close" />
```
so that the user can potentially provide their own Icons. The button just renders the provided Icon with a `scale` prop.

Reply:
That's better than.